panther_hardware_interfaces/test/unit/panther_system/robot_driver/test_roboteq_robot_driver.cpp

```diff
@@ -0,0 +1,569 @@
+    f_battery_current_1 + f_battery_current_2);
+
+  EXPECT_EQ(static_cast<std::int16_t>(rear_driver_state.GetTemperature()), r_temp);
+  EXPECT_EQ(static_cast<std::int16_t>(rear_driver_state.GetHeatsinkTemperature()), r_heatsink_temp);
+  EXPECT_EQ(static_cast<std::uint16_t>(rear_driver_state.GetVoltage() * 10.0), r_volt);
+  EXPECT_EQ(
+    static_cast<std::int16_t>(rear_driver_state.GetCurrent() * 10.0),
+    r_battery_current_1 + r_battery_current_2);
+
+  EXPECT_TRUE(front_data.GetFaultFlag().GetMessage().overheat);
+  EXPECT_TRUE(front_data.GetScriptFlag().GetMessage().encoder_disconnected);
+  EXPECT_TRUE(front_data.GetRuntimeError(MotorChannels::RIGHT).GetMessage().loop_error);
+  EXPECT_TRUE(front_data.GetRuntimeError(MotorChannels::LEFT).GetMessage().safety_stop_active);
+
+  EXPECT_TRUE(rear_data.GetFaultFlag().GetMessage().overvoltage);
+  EXPECT_TRUE(rear_data.GetScriptFlag().GetMessage().amp_limiter);
+  EXPECT_TRUE(rear_data.GetRuntimeError(MotorChannels::RIGHT).GetMessage().forward_limit_triggered);
+  EXPECT_TRUE(rear_data.GetRuntimeError(MotorChannels::LEFT).GetMessage().reverse_limit_triggered);
+}
+
+TEST_F(TestPantherRobotDriver, UpdateDriverStateTimestamps)
+{
+  auto read_driver_state_method = []() {
+    panther_hardware_interfaces::DriverState state;
+    clock_gettime(CLOCK_MONOTONIC, &state.flags_current_timestamp);
+    clock_gettime(CLOCK_MONOTONIC, &state.voltages_temps_timestamp);
+    return state;
+  };
+
+  ON_CALL(*robot_driver_->GetMockFrontDriver(), ReadDriverState())
+    .WillByDefault(::testing::Invoke(read_driver_state_method));
+  ON_CALL(*robot_driver_->GetMockRearDriver(), ReadDriverState())
+    .WillByDefault(::testing::Invoke(read_driver_state_method));
+
+  robot_driver_->UpdateDriversState();
+
+  // sleep for timeout and check if timestamps were updated correctly
+  std::this_thread::sleep_for(
+    panther_hardware_interfaces_test::kCANopenSettings.pdo_driver_state_timeout_ms +
+    std::chrono::milliseconds(10));
+
+  robot_driver_->UpdateDriversState();
+
+  EXPECT_FALSE(robot_driver_->GetData(RoboteqRobotDriverWrapper::kFrontDriverName)
+                 .IsDriverStateDataTimedOut());
+  EXPECT_FALSE(
+    robot_driver_->GetData(RoboteqRobotDriverWrapper::kRearDriverName).IsDriverStateDataTimedOut());
+}
+
+TEST_F(TestPantherRobotDriver, UpdateDriverStateTimeout)
+{
+  timespec current_time;
+  clock_gettime(CLOCK_MONOTONIC, &current_time);
+
+  panther_hardware_interfaces::DriverState state = {
+    0, 0, 0, 0, 0, 0, 0, 0, 0, current_time, current_time};
+
+  ON_CALL(*robot_driver_->GetMockFrontDriver(), ReadDriverState())
+    .WillByDefault(::testing::Return(state));
+  ON_CALL(*robot_driver_->GetMockRearDriver(), ReadDriverState())
+    .WillByDefault(::testing::Return(state));
+
+  // sleep for pdo_driver_state_timeout_ms + 10ms
+  std::this_thread::sleep_for(
+    panther_hardware_interfaces_test::kCANopenSettings.pdo_driver_state_timeout_ms +
+    std::chrono::milliseconds(10));
```

Comment:
Applies to all similar cases:

How about setting deprecated current time (`time.now() - timeout`) instead of sleeping in the test case? This'll speed up tests execution.